jar.go

```diff
@@ -295,6 +299,60 @@ func (j *Jar) cookies(u *url.URL, now time.Time) (cookies []*http.Cookie) {
 	return cookies
 }
 
+// AllCookies returns all cookies in the jar. The returned cookies will
+// have Domain, Expires, HttpOnly, Name, Secure, Path, and Value filled
+// out. Any stored but expired cookies will not be returned. This
+// function does not modify the cookie jar.
+func (j *Jar) AllCookies() (cookies []*http.Cookie) {
+	return j.allCookies(time.Now())
+}
+
+// allCookies is like AllCookies but takes the current time as a parameter.
+func (j *Jar) allCookies(now time.Time) []*http.Cookie {
+	var selected []entry
+	j.mu.Lock()
+	defer j.mu.Unlock()
+	for _, submap := range j.entries {
+		for _, e := range submap {
+			if !e.Expires.After(now) {
+				// Do not return expired cookies.
+				continue
+			}
+			selected = append(selected, e)
+		}
+	}
+
+	sort.Sort(byCanonicalHost{byPathLength(selected)})
+	cookies := make([]*http.Cookie, len(selected))
+	for i, e := range selected {
+		cookies[i] = &http.Cookie{
+			Name:     e.Name,
+			Value:    e.Value,
+			Path:     e.Path,
+			Domain:   e.Domain,
```

Comment:
Unfortunately this is not the same as the domain that was originally used to create the cookie, and that means that AFAICS we can't losslessly get a cookie and then set it back again.

Perhaps add a comment to that effect and a TODO to the code?